sunpy/net/dataretriever/client.py

```diff
@@ -207,7 +227,42 @@ def _get_url_for_timerange(cls, timerange, **kwargs):
              attributes passed to
              `~sunpy.net.dataretriever.client.GenericClient.search`.
         """
-        raise NotImplementedError
+        if cls.baseurl is None or cls.extractor is None:
+            raise NotImplementedError
+        matchdict = cls._get_matchdict(**kwargs)
+        scraper = Scraper(cls.baseurl, regex=True, extractor=cls.extractor, matcher=matchdict)
+        return scraper.filelist(timerange)
+
+    @classmethod
+    def _get_matchdict(cls, **kwargs):
+        """
+        Method which generates a dict representing all valid attributes
+        that can be retrieved from the file url.
+
+        Parameters
+        ----------
+        \\*\\*kwargs: `dict`
+             Any extra keywords to refine the search. Generated from the
+             attributes passed to
+             `~sunpy.net.dataretriever.client.GenericClient.search`.
+        """
+        a = cls.register_values()
```

Comment:
I wonder how many times we call this and is it worth storing as a dict on the class itself.